parser/src/aspects/function_declaration.rs

```diff
@@ -32,33 +32,57 @@ impl<'a> FunctionDeclarationAspect<'a> for Parser<'a> {
             )?
             .value;
 
+        let segment_start = self.cursor.relative_pos(fun).start;
+
         //consume blanks for each function declaration components
         self.cursor.advance(blanks());
         let name = self.parse_fn_declaration_name()?;
         self.cursor.advance(blanks());
         let tparams = self.parse_type_parameter_list()?.0;
         self.cursor.advance(blanks());
         let params = self.parse_fn_parameter_list()?;
-        self.cursor.advance(blanks());
+
+        if let Some(token) = self.cursor.lookahead(blanks().then(of_type(SemiColon))) {
+            return Ok(FunctionDeclaration {
+                name,
+                type_parameters: tparams,
+                parameters: params,
+                return_type: None,
+                body: None,
+                segment: segment_start..self.cursor.relative_pos_ctx(token).end,
+            });
+        }
+
         let rtype = self.parse_fn_return_type()?;
-        self.cursor.advance(blanks());
+
+        if let Some(token) = self.cursor.lookahead(blanks().then(of_type(SemiColon))) {
+            return Ok(FunctionDeclaration {
+                name,
+                type_parameters: tparams,
+                parameters: params,
+                return_type: rtype,
+                body: None,
+                segment: segment_start..self.cursor.relative_pos_ctx(token).end,
+            });
+        }
 
         let body = self
             .cursor
-            .force(blanks().then(of_type(Equal)), "expected '='")
+            .force(blanks().then(of_type(Equal)), "`=` expected")
```

Comment:
```suggestion
            .force(blanks().then(of_type(Equal)), "expected '='")
```
You expect something, not someone.